crates/extensions/c8y_mapper_ext/src/inventory.rs

```diff
@@ -467,29 +467,19 @@ mod tests {
             r#"{"name":"firmware", "version":"1.0"}"#,
         );
 
+        converter
+            .try_register_source_entities(&twin_message)
+            .await
+            .unwrap();
```

Comment:
Instead of repeating the validation of the auto-registered messages in the conversion tests for all data types: measurements, events, twin updates etc, I just wrote two new tests dedicated for auto-registration named `auto_register_child_device` and `auto_register_child_device_service` where the registration messages are validated. One additional tweak that I can make to those tests is to make it `test_case` based and add variants for all the data types as additional test cases.